### PR TITLE
MC-10842 : Added swift api documentation draft

### PR DESCRIPTION
**Pull request by fyrnaga**

### CODE REVIEW

File: source/includes/swift/_containers.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+Attributes | &nbsp;
+------- | -----------
+`id` <br/>*string* | The id of the container 
+`name` <br/>*string* | The name of the container
+`isPublic`<br/>*boolean* | Specifies if the container is public or not
+`displaySize`<br/>*Object* | Object containing the information to display the size in the UI.
+`size`<br/>*string* | The size in bytes of the container
+`objectCount`<br/>*string* | The number of object which are part of the container
+`accessType`<br/>*string* | The access type of the container. It can either be public or private
+`locked`<br/>*string* | Identify if the container is locked or not.
```
Comment (lamminade) [inline source/includes/swift/_containers.md]: This is a little confusing, is this supposed to be a boolean as in the above example?
if it is, maybe something like "Specifies if the container is locked or not"

File: source/includes/swift/_containers.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+}
+```
+
+<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/buckets</code>
+
+Create a new container
+
+Required | &nbsp;∂===
+------- | -----------
+`name` <br/>*string* | The name of the container. It cannot exceed 256 characters and cannot contain '/'
```
Comment (lamminade) [inline source/includes/swift/_containers.md]: Do we want to end this with a period for consistency

File: source/includes/swift/_containers.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+```json
+{
+  "name": "newContainerName"
+}
+```
+
+<code>POST /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/buckets/:id?operation=rename</code>
+
+<aside class="notice">
+Renaming does not exists in SWIFT. It consists of a creation of a new container, a copy of all objects in this new container and the delete of the old container.
```
Comment (lamminade) [inline source/includes/swift/_containers.md]:
```suggestion
Renaming does not exist in SWIFT. It instead consists of the creation of a new container, a copy of all objects into this new container, and the deletion of the old container.
```

File: source/includes/swift/_objects.md [hunk 4]
```diff
@@ -0,0 +1,240 @@
...
+}
+```
+
+<code>GET /services/<a href="#administration-service-connections">:service_code</a>/<a href="#administration-environments">:environment_name</a>/objects?bucket_name=:bucket_name</code>
+
+Retrieve a list of all objects in a given [environment](#administration-environments) in the specified container
+
+Mandatory | &nbsp;
+------- | -----------
+`bucket_name` <br/>*string* | The name of the bucket from which the object must be fetched.
```
Comment (lamminade) [inline source/includes/swift/_objects.md]: Again some more inconsistency here and in the attributes with which sentences end in a period or not

**Review by lamminade — APPROVED** [review]
Some small changes (i know this is a draft) but looks good!

File: source/includes/swift/_objects.md (hunk 4 above)
Comment (fyrnaga) [reply source/includes/swift/_objects.md]: Will fix

File: source/includes/swift/_containers.md (hunk 1 above)
Comment (fyrnaga) [reply source/includes/swift/_containers.md]: forgot to change it. will do

File: source/includes/swift/_containers.md (hunk 2 above)
Comment (fyrnaga) [reply source/includes/swift/_containers.md]: will fix

File: source/includes/swift/_containers.md (hunk 3 above)
Comment (fyrnaga) [reply source/includes/swift/_containers.md]: will apply. thx

File: source/includes/swift/_containers.md (hunk 1 above)
Comment (fyrnaga) [reply source/includes/swift/_containers.md]: f47db080986d4eed0065f2b048c24c5faeafb7be

File: source/includes/swift/_containers.md (hunk 2 above)
Comment (fyrnaga) [reply source/includes/swift/_containers.md]: f47db080986d4eed0065f2b048c24c5faeafb7be

File: source/includes/swift/_containers.md (hunk 3 above)
Comment (fyrnaga) [reply source/includes/swift/_containers.md]: f47db080986d4eed0065f2b048c24c5faeafb7be

File: source/includes/swift/_objects.md (hunk 4 above)
Comment (fyrnaga) [reply source/includes/swift/_objects.md]: f47db080986d4eed0065f2b048c24c5faeafb7be

File: source/includes/swift/_objects.md [hunk 5]
```diff
@@ -0,0 +1,262 @@
...
+------- | -----------
+`id` <br/>*string* | The id of the object .
+`name` <br/>*string* | The name of the object.
+`displayName`<br/>*string* | The name rendered for display.
+`path`<br/>*string* | The path of the object.
+`size`<br/>*int* | The size in bytes of the object. It is 0 for object that are used as folders (Generated by UI).
+`isNode`<br/>*boolean* | The access type of the container. It can either be public or private.
+`bucketName`<br/>*string* | The name of the bucket that the object is part of.
+`displaySize`<br/>*object* | Object containing the information to display the size in the UI.
+`contentType`<br/>*string* | The mime type corresponding to the content of the file.
```
Comment (GuillaumeVialle) [inline source/includes/swift/_objects.md]: Can you capitalize MIME?

File: source/includes/swift/_objects.md [hunk 6]
```diff
@@ -0,0 +1,262 @@
...
```
Comment (GuillaumeVialle) [inline source/includes/swift/_objects.md]: same here

File: source/includes/swift/_objects.md [hunk 7]
```diff
@@ -0,0 +1,262 @@
...
+`lastModified`<br/>*string* | The last date that the file was modified.
+
+
+<!-------------------- DOWNLOAD AN OBJECT -------------------->
+
+#### Download an object (To complete)
```
Comment (GuillaumeVialle) [inline source/includes/swift/_objects.md]: `(To complete)` Should we merge something that isn't implemented yet?

Comment (fyrnaga) [reply source/includes/swift/_objects.md]: good point. @simongodard what is your take on this?

File: source/includes/swift/_objects.md (hunk 6 above)
Comment (fyrnaga) [reply source/includes/swift/_objects.md]: cdaa1dc4734e0146a9f3a346d80e276c4993bfbe

File: source/includes/swift/_objects.md (hunk 5 above)
Comment (fyrnaga) [reply source/includes/swift/_objects.md]: cdaa1dc4734e0146a9f3a346d80e276c4993bfbe

File: source/includes/swift/_objects.md (hunk 7 above)
Comment (simongodard) [reply source/includes/swift/_objects.md]: We should only merge completed doc. I would remove that section if not completed.

Comment (fyrnaga) [reply source/includes/swift/_objects.md]: Should i keep only the container since the objects needs some fixes?

Comment (simongodard) [reply source/includes/swift/_objects.md]: Yes

Comment (fyrnaga) [reply source/includes/swift/_objects.md]: will do

Comment (fyrnaga) [reply source/includes/swift/_objects.md]: d1a1327e888bdf3d2f16d91831f2c29d5a3e36f6

**Review by GuillaumeVialle — APPROVED** [review]
Thanks  for the changes

**Pull request merged**